app/policies/project_policy.rb

```diff
@@ -16,4 +16,22 @@ def new?
   def create?
     user.present?
   end
+
+  def edit?
+    update?
+  end
+
+  def update?
+    return false if user.nil?
+
+    membership = ProjectMembership.find_by(project: record, user: user)
+    membership.present? && (membership.member? || membership.owner?)
+  end
+
+  def destroy?
+    return false if user.nil?
+
+    membership = ProjectMembership.find_by(project: record, user: user)
+    membership.present? && membership.owner?
```

Comment:
How about
```suggestion
    ProjectMembership.find_by(project: record, user: user, role: :owner).present?
```

Reply:
And move finding to methods `member?` and `owner?` and in `destroy?` just check `user.present? && owner?`